functions/levante-admin/src/upsertAdministration.ts

```diff
@@ -50,6 +71,251 @@ interface IAdministrationDoc {
   creatorName: string;
 }
 
+const createAssignments = async (
+  administrationId: string,
+  administrationDocRef: DocumentReference,
+  currData: IAdministration,
+  creatorUid?: string,
+  isNewAdministration: boolean = false
+) => {
+  const { minimalOrgs } = await standardizeAdministrationOrgs({
+    administrationId,
+    administrationDocRef,
+    currData,
+    copyToSubCollections: true,
+    forceCopy: true,
+  });
+
+  const orgChunks = chunkOrgs(minimalOrgs, 100);
+  const allCreatedUserIds: string[] = [];
+  const processedChunks: Array<{ orgChunk: IOrgsList; userIds: string[] }> = [];
+
+  try {
+    const chunkResults = await Promise.all(
+      orgChunks.map(async (orgChunk, i) => {
+        try {
+          const result = await updateAssignmentsForOrgChunkHandler({
+            administrationId,
+            administrationData: currData,
+            orgChunk,
+            mode: "add",
+          });
+          return { orgChunk, result, chunkIndex: i };
+        } catch (error: any) {
+          const err = error instanceof Error ? error : new Error(String(error));
+          return {
+            orgChunk,
+            result: { success: false, userIds: [], error: err },
+            chunkIndex: i,
+          };
+        }
+      })
+    );
+
+    for (const { orgChunk, result } of chunkResults) {
+      if (result.success) {
+        allCreatedUserIds.push(...result.userIds);
```

Comment:
Do we need to check if it's successful, won't the `promise.all` just reject at the first error anyways?